infra/terraform/global/assets/create_etcd_ebs_snapshot/create_etcd_ebs_snapshot.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/aws"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"os"
+	"log"
+)
+
+var (
+	connection = ec2.New(session.New())
+)
+
+
+func main() {
+
+	instanceKeyFilter   := os.Getenv("INSTANCE_KEY")
+	instanceValueFilter := os.Getenv("INSTANCE_VALUE")
+
+	// Filter instances with role set to master
+	instanceInput := &ec2.DescribeInstancesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name: 	aws.String(fmt.Sprintf("tag:%s", instanceKeyFilter )),
+				Values: []*string{ aws.String(instanceValueFilter)},
+			},
+		},
+	}
+
+	// A reservation corresponds to a command to start instances
+	// A reservation is what you do to provision instances, while an instance is what you get
+	reservation, err := connection.DescribeInstances(instanceInput)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	// Iterate over reservations
+	for index := range reservation.Reservations {
```

Comment:
My Go is rusty (and it wasn't good anyway in the first place) so double check, but as far as I can tell, you don't care much about the `index` here right? You're only using it to loop through the reservations array.

Maybe you could rename `reservation` to `results` (I know...) and then the loop would look like:

```Go
results, err := // ...
for _, reservation in results.Reservations {
  for _, instance := range reservation.Instances {
    for _, block := range inst.BlockDeviceMappings {
      // ...
    }
  }
}
```

Or something like that. It's pretty much the same but you reduce the confusion regarding what variable is actually containing the reservation, etc...